test/common/cro_cr_success.go

```diff
@@ -0,0 +1,128 @@
+	}
+
+	for _, postGresName := range postGresToCheck {
+
+		strategy := getResourceStrategy(t, ctx)
+		// AMQAuthService postgres is always in cluster
+		if postGresName == constants.AMQAuthServicePostgres {
+			strategy = openShiftProvider
+		}
+
+		requestBody, _ := getCustomResourceJson(t, ctx, "postgres", postGresName)
+
+		postGres := &crov1.Postgres{}
+		err := json.Unmarshal(requestBody, postGres)
+
+		if err != nil {
+			t.Fatalf("Failed to unmarshall json: %s", err)
+		}
+
+		if postGres.Status.Phase != croTypes.PhaseComplete && postGres.Status.Strategy != strategy {
+			t.Fatalf("%s Postgres not ready with phase: %s, message: %s, provider, %s", postGresName, postGres.Status.Phase, postGres.Status.Message, postGres.Status.Provider)
+		}
+	}
+}
+
+func TestCRORedisSuccessfulState(t *testing.T, ctx *TestingContext) {
+	redisToCheck := []string{
+		fmt.Sprintf("%s%s", constants.ThreeScaleBackendRedisPrefix, InstallationName),
+		fmt.Sprintf("%s%s", constants.ThreeScaleSystemRedisPrefix, InstallationName),
+	}
+
+	strategy := getResourceStrategy(t, ctx)
+
+	for _, redisName := range redisToCheck {
+		requestBody, _ := getCustomResourceJson(t, ctx, "redis", redisName)
+
+		redis := &crov1.Redis{}
+		err := json.Unmarshal(requestBody, redis)
+
+		if err != nil {
+			t.Fatalf("Failed to unmarshall json: %s", err)
+		}
+
+		if redis.Status.Phase != croTypes.PhaseComplete && redis.Status.Strategy != strategy {
+			t.Fatalf("%s redis not ready with phase: %s, message: %s, provider, %s", redisName, redis.Status.Phase, redis.Status.Message, redis.Status.Provider)
+		}
+	}
+}
+
+func TestCROBlobStorageSuccessfulState(t *testing.T, ctx *TestingContext) {
+	blobStorageToCheck := []string{
+		fmt.Sprintf("backups-blobstorage-%s", InstallationName),
+		fmt.Sprintf("threescale-blobstorage-%s", InstallationName),
+	}
+
+	strategy := getResourceStrategy(t, ctx)
+
+	for _, blobStorageName := range blobStorageToCheck {
+		requestBody, _ := getCustomResourceJson(t, ctx, "blobstorages", blobStorageName)
+
+		blobStorage := &crov1.BlobStorage{}
+		err := json.Unmarshal(requestBody, blobStorage)
+
+		if err != nil {
+			t.Fatalf("Failed to unmarshall json: %s", err)
+		}
+
+		if blobStorage.Status.Phase != croTypes.PhaseComplete && blobStorage.Status.Strategy != strategy {
+			t.Fatalf("%s blob storage not ready with phase: %s, message: %s, provider, %s", blobStorageName, blobStorage.Status.Phase, blobStorage.Status.Message, blobStorage.Status.Provider)
+		}
+	}
+}
+
+// Function to get a custom resource json without needing to depend on operator-sdk
+func getCustomResourceJson(t *testing.T, ctx *TestingContext, resource string, resourceName string) ([]byte, error) {
+	request := ctx.ExtensionClient.RESTClient().Get().Resource(resource).Name(resourceName).Namespace(requestNameSpace).RequestURI(requestUrl).Do()
+	requestBody, err := request.Raw()
+
+	if err != nil {
+		t.Fatalf("failed to get %s resource %s: %s", resource, resourceName, err)
```

Comment:
Sorry, me again bothering with the style :), wouldn't be better to return the error instead of passing `t`

Reply:
Yup, sounds good